flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -697,6 +697,7 @@ private int calculateIncreasingSelfState(int ignoreBranches, int takeBranches) {
 
 		final Stack<State<T>> states = new Stack<>();
 		states.push(state);
+		ConditionContext<T> context = new ConditionContext<T>(this, sharedBuffer, computationState);
```

Comment:
Rather than `computationState` just pass `conditionContext` to the `createDecisionGraph` method

Reply:
I think it needs the `conditionContext` and `computationState` and should replace the `sharedBuffer` with `conditionContext`.